server/journeys/referral/check-answers/controller.ts

```diff
@@ -20,6 +22,51 @@ export class ReferralCheckAnswersController {
     )
   }
 
+  checkSubmitToAPI: RequestHandler = async (req, res, next) => {
+    const prisoner = req.journeyData.prisoner!
+    const referral = req.journeyData.referral!
+    try {
+      await this.createReferral(req, {
+        logCode: prisoner.prisonId,
+        referral: {
+          contributoryFactors: referral.contributoryFactors!.map(factor => ({
+            factorTypeCode: factor.factorType.code,
+            ...(factor.comment !== undefined && { comment: factor.comment }),
```

Comment:
This is kinda complex, but if we're just not setting `comment` on the object if `factor.comment` is undefined, why do we need this check? afaik setting something to undefined is virtually identical to not setting it?